frontend/src/components/DataTable.vue

```diff
@@ -1,29 +1,66 @@
 <template>
-  <div class="flex justify-end gap-2 mb-2">
-    <SearchField
-      v-if="config.search"
-      v-model="search"
-    />
-    <label
-      v-for="filter in filters"
-      :key="filter"
-    >
-      <select
-        v-model="filter.value"
-        class="rounded-lg border-0"
-        name="type"
+  <div class="grid grid-cols-2 gap-2">
+    <div class="flex justify-start gap-2 mb-2">
+      <label
+        v-for="[ key, filter ] in Object.entries(filters).filter(([key , filter]) => filter.align === 'left')"
+        :key="key"
+      >
+        <slot
+          :key="key"
+          :filter="filter"
+          :name="`filters(${key})`"
+          :rows="rowsInternal"
+        >
+          <!--  @todo Move to sub-component-->
+          <select
+            v-model="filter.value"
+            class="rounded-lg border-0"
+            name="type"
+          >
+            <option
+              v-for="option in filter.options"
+              :key="option.value"
+              :value="option.value"
+            >
+              {{ option.label }}
+            </option>
+          </select>
+        </slot>
+      </label>
+    </div>
+    <div class="flex justify-end gap-2 mb-2">
```

Comment:
I'm sure we can avoid duplication here, but i did not have time to try

Reply:
When left and right are simple divs, this could works by using v-for="i in 2" and then v-if="(i === 1 && filter.align === 'left') || ...